develop/toolkit/api-libraries/py-substrate-interface.md

```diff
@@ -1,4 +1,72 @@
 ---
 title: Python Substrate Interface
-description: TODO
----
\ No newline at end of file
+description: Learn how to connect to Polkadot SDK-based nodes, query data, submit transactions, and manage blockchain interactions using the Python Substrate Interface.
+---
+
+# Python Substrate Interface
+
+## Introduction
+
+This Python library specializes in interfacing with a Polkadot SDK-based node, querying storage, composing extrinsics, SCALE encoding/decoding, and providing additional convenience methods to deal with the features and metadata of the Substrate runtime.
+
+The Python Substrate Interface is a powerful library that enables interaction with Polkadot SDK-based chains. It provides essential functionality for:
+
+- Querying on-chain storage
+- Composing and submitting extrinsics
+- SCALE encoding/decoding
+- Interacting with Substrate runtime metadata
+- Managing blockchain interactions through convenient utility methods
+
+## Installation
+
+Install the library using pip:
+
+```py
+pip install substrate-interface
+```
+
+## Get Started
+
+This guide will walk you through the basic operations with the Python Substrate Interface: connecting to a node, reading chain state, and submitting transactions.
+
+### Establishing Connection
+
+The first step is to establish a connection to a Polkadot SDK-based node. You can connect to either a local or remote node:
+
+```py
+--8<-- 'code/develop/toolkit/api-libraries/py-substrate-interface/substrate_interface.py'
+```
+
+### Reading Chain State
+
+You can query various on-chain storage items. Here's how to check an account's balance and other details:
```

Comment:
Please add a sentence explaining that each query requires the specification of the pallet name, the storage item, and the required parameters 

So then adding this specific snippet would have context